packages/arb-token-bridge-ui/src/util/deposits/helpers.ts

```diff
@@ -265,6 +298,149 @@ const updateClassicDepositStatusData = async ({
   return completeDepositTx
 }
 
+export async function updateTeleporterDepositStatusData({
+  assetType,
+  parentChainId,
+  childChainId,
+  txID
+}: {
+  assetType: AssetType
+  parentChainId: number
+  childChainId: number
+  txID: string
+}): Promise<{
+  status?: TxnStatus
+  timestampResolved?: string
+  l1ToL2MsgData?: L1ToL2MessageData
+  l2ToL3MsgData?: L2ToL3MessageData
+}> {
+  const isNativeCurrencyTransfer = assetType === AssetType.ETH
+
+  const sourceChainProvider = getProvider(parentChainId)
```

Comment:
Is it possible to do any sanity checks to make sure it's deposit? What if the method is called incorrectly and `parentChainId` is the destination (withdrawal case). In that case we could throw some that it goes to catch and returns the default state.